frontend/src/loglist/__tests__/LogsTableWrapper.test.jsx

```diff
@@ -10,26 +10,46 @@ Enzyme.configure({ adapter: new Adapter() });
 
 describe("<LogsTableWrapper/> Class Component", () => {
   let wrapper;
-  const dummyValue = [new LogInfo(1, null, "Info", "Log 1 info")];
+  const dummyLog = [new LogInfo(1, null, "Info", "Log 1 info")];
   const dummySource = {
     getAllLogs() {
-      return new Promise((resolve) => resolve(dummyValue));
+      return new Promise((resolve) => resolve(dummyLog));
     }
   };
-  describe("render() function", () => {
+
+  beforeEach(() => {
     wrapper = Enzyme.shallow(<LogsTableWrapper logsDataSource={dummySource} />);
+  });
+
+  afterEach(() => {
+    wrapper.unmount();
+  });
+
+  describe("render() function", () => {
     describe("returns a component that", () => {
-      it("Contains 1 <LogsTable/> component", () => {
-        expect(wrapper.find(LogsTable)).toHaveLength(1);
+      it("Contains 1 <LogsTable/> component with expected props", () => {
+        const logsTable = wrapper.find(LogsTable);
+        expect(logsTable).toHaveLength(1);
+
+        const logsTableProps = logsTable.props();
+        const expected = wrapper.instance().getColumnInfo();
+        expect(logsTableProps.columns).toEqual(expected);
+      });
+    });
+  });
+
+  describe("componentDidMount() function", () => {
+    describe("calls LogApi getAllLogs() with the expected arguments", () => {
+      it("then sets the state to resolved value", () => {
```

Comment:
1) test name/structure could be better. You can find other tests i've written recently for componentDidMount() [here](https://github.com/bean-pod/switchboard/blob/0ac2f251e590768ca9c5f5985f69d64177162180/frontend/src/devicelist/__tests__/DeviceListPageContents.test.jsx) for an example
2) this test should be checking that componentDidMount() calls LogsSource.getAllLogs. The way it's written now, you're testing LogApi.getAllLogs and handleLogsChange() instead of componentDidMount. Also, because you aren't explicitly setting a state beforehand, you aren't necessarily checking that it changed the state, 
3) component did mount is a special case and while, yes, it is called as part of the lifecycle functions, you can disable the lifecycle functions are explicitly call it here. See previously linked file.

Reply:
370581c